Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs

```
@@ -584,7 +584,12 @@ private async Task RefreshDatabaseAccountInternalAsync(bool forceRefresh)
             {
                 this.LastBackgroundRefreshUtc = DateTime.UtcNow;
                 this.locationCache.OnDatabaseAccountRead(await this.GetDatabaseAccountAsync(true));
-
+            }

```

[Not blocker]
This will fix the reported issue. 

From the call stack shared in the issue, it sees the real issue is AggregationException thrown, not un-observed exception? We might only be fixing one possible occurance?

---

Basically the catch block inside the `RefreshDatabaseAccountInternalAsync()` method should cover any exception that is thrown from the `GetDatabaseAccountAsync()`.

```
    catch (Exception ex)
    {
        DefaultTrace.TraceWarning("Failed to refresh database account with exception: {0}. Activity Id: '{1}'",
            ex,
            System.Diagnostics.Trace.CorrelationManager.ActivityId);
    }
```

Earlier, the exception was not handled and was thrown from this point, which was creating the noise.

---

Who are the callers and how is it unhandled?

---

Well, there are 2 places we do invoke this method from.

- First, the `StartLocationBackgroundRefreshLoop()` which creates a loop to continuously do background refresh every 5 minutes.
- Second, from the `ClientRetryPolicy.ShouldRetryOnEndpointFailureAsync()` which invokes the `RefreshDatabaseAccountInternalAsync()` internally.

Now, the case the dataverse team encountered (from the issue #3994 ) is the second one. When I look around the code, I do not see any exception handling for a case where `RefreshDatabaseAccountInternalAsync()` method throws any runtime exception, unless I am missing something.